File: packages/typescript/tests/staging.smoke.test.ts
import { describe, expect, it } from "vitest";
import { NeuroVerifyClient } from "../src/index.js";

const apiKey = process.env.NEURALDEFEND_STAGING_API_KEY;
const imagePath = process.env.NEURALDEFEND_STAGING_IMAGE;
const videoPath = process.env.NEURALDEFEND_STAGING_VIDEO;
const enabled = Boolean(apiKey && imagePath && videoPath);

describe.skipIf(!enabled)("staging contract", () => {
  it("returns a consistent image result", async () => {
    // Use the SDK retry policy for HTTP 429/500/503. Staging smoke previously
    // set maxRetries: 0, so a single transient 503 failed the weekly job.
    const client = NeuroVerifyClient.staging({
      apiKey: apiKey!,
    });
    const result = await client.detectImage(imagePath!);

    expect(["success", "rejected"]).toContain(result.status);
    expect(result.uniqueTrxId).toBeTruthy();
    expect(typeof result.billable).toBe("boolean");
    expect(result.scored).toBe(result.status === "success");
    expect(result.rejected).toBe(result.status === "rejected");
  });

  it("returns a consistent video result", async () => {
    const client = NeuroVerifyClient.staging({
      apiKey: apiKey!,
    });
    const result = await client.detectVideo(videoPath!, { maxFrames: 2 });

    expect(["success", "rejected"]).toContain(result.status);
    expect(result.uniqueTrxId).toBeTruthy();
    expect(typeof result.billable).toBe("boolean");
    expect(result.scored).toBe(result.status === "success");
    expect(result.rejected).toBe(result.status === "rejected");
    if (result.status === "success") {
      expect(result.videoRiskScore).not.toBeNull();
      expect(["low", "medium", "high"]).toContain(result.videoRiskLevel);
    }
  });
});

File: packages/java/src/test/java/com/neuraldefend/StagingSmokeTest.java
package com.neuraldefend;

import java.nio.file.Files;
import java.nio.file.Path;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;

import static org.junit.jupiter.api.Assertions.*;

final class StagingSmokeTest {
    @Test
    @EnabledIfEnvironmentVariable(named = "NEURALDEFEND_STAGING_API_KEY", matches = ".+")
    void stagingImageContract() throws Exception {
        String apiKey = System.getenv("NEURALDEFEND_STAGING_API_KEY");
        Path path = stagingFixture("NEURALDEFEND_STAGING_IMAGE");

        ClientOptions options = new ClientOptions();
        options.apiKey = apiKey;
        options.maxRetries = 0;
        NeuroVerifyClient client = NeuroVerifyClient.staging(options);

        ImageResult result = client.detectImage(Media.fileMedia(path));
        assertConsistentResult(
                result.status, result.uniqueTrxId, result.billable, result.scored(), result.rejected());
    }

    @Test
    @EnabledIfEnvironmentVariable(named = "NEURALDEFEND_STAGING_API_KEY", matches = ".+")
    void stagingVideoContract() throws Exception {
        String apiKey = System.getenv("NEURALDEFEND_STAGING_API_KEY");
        Path path = stagingFixture("NEURALDEFEND_STAGING_VIDEO");

        ClientOptions options = new ClientOptions();
        options.apiKey = apiKey;
        options.maxRetries = 0;
        NeuroVerifyClient client = NeuroVerifyClient.staging(options);

        VideoOptions videoOptions = new VideoOptions();
        videoOptions.maxFrames = 2;
        VideoResult result = client.detectVideo(Media.fileMedia(path), videoOptions);
        assertConsistentResult(
                result.status,
                result.uniqueTrxId,
                result.billable,
                result.scored(),
                result.rejected());
        if (result.scored()) {
            assertNotNull(result.videoRiskScore);
            assertNotNull(result.videoRiskLevel);
            assertTrue(
                    switch (result.videoRiskLevel) {
                        case "low", "medium", "high" -> true;
                        default -> false;
                    });
        }
    }

    private static Path stagingFixture(String envName) {
        String value = System.getenv(envName);
        if (value == null || value.isBlank()) {
            fail(envName + " is not configured");
        }
        Path path = Path.of(value).toAbsolutePath().normalize();
        if (!Files.isRegularFile(path)) {
            fail(envName + " does not identify a staging fixture");
        }
        return path;
    }

    private static void assertConsistentResult(
            String status, String trxId, boolean billable, boolean scored, boolean rejected) {
        assertTrue("success".equals(status) || "rejected".equals(status), "unexpected status");
        assertFalse(trxId.isBlank());
        assertEquals("success".equals(status), scored);
        assertEquals("rejected".equals(status), rejected);
    }
}
